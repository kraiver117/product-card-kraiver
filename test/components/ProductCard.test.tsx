import React from "react";
import renderer, { act } from 'react-test-renderer';
import ProductCard from "../../src/components";
import { product2 } from '../data/products';

describe('test product card functionality', () => { 
  test('should show the component correctly', () => {
    // Wrapper to create a renderer component
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {
          ({ product }) => (
            <>
              <h1>{product.title}</h1>
            </>
          )
        }
      </ProductCard>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should increase count', () => {  
    const wrapper =  renderer.create(
      <ProductCard product={product2}>
        {
          ({ count, increaseBy }) => (
            <>
              <h1>Product Card</h1>
              <span>{ count }</span>
              <button onClick={() => increaseBy(1)}></button>
            </>
          )
        }
      </ProductCard>
    );

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    // Action - click the button
    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();
    
    //Check counter value
    expect((tree as any).children[1].children[0]).toBe('1');
  })
});