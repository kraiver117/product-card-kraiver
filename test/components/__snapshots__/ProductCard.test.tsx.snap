// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test product card functionality should increase count 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`test product card functionality should show the component correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Coffe Mug
  </h1>
</div>
`;
